src/actions/query.js

```diff
@@ -183,37 +194,30 @@ function msearch(
 			}
 
 			orderOfQueries.forEach((component, index) => {
-				const response = res.responses[index];
-				const { timestamp } = getState();
-
-				if (timestamp[component] === undefined || timestamp[component] < res._timestamp) {
-					if (response.hits) {
-						dispatch(setTimestamp(component, res._timestamp));
-						dispatch(updateHits(component, response.hits, response.took, appendToHits));
-						dispatch(setLoading(component, false));
-					}
+				handleTransformResponse(res.responses[index], component).then((response) => {
+					const { timestamp } = getState();
+					if (
+						timestamp[component] === undefined
+						|| timestamp[component] < res._timestamp
+					) {
+						if (response.hits) {
+							dispatch(setTimestamp(component, res._timestamp));
+							dispatch(updateHits(component, response.hits, response.took, appendToHits));
+							dispatch(setLoading(component, false));
+						}
 
-					if (response.aggregations) {
-						dispatch(updateAggs(component, response.aggregations, appendToAggs));
+						if (response.aggregations) {
+							dispatch(updateAggs(component, response.aggregations, appendToAggs));
+						}
 					}
-				}
-			});
-		};
-
-		const handleError = (error) => {
-			console.error(error);
-			orderOfQueries.forEach((component) => {
-				if (queryListener[component] && queryListener[component].onError) {
-					queryListener[component].onError(error);
-				}
-				dispatch(setError(component, error));
-				dispatch(setLoading(component, false));
+				}).catch((err) => {
+					handleError(err);
```

Comment:
This doesn't seem right 🤔 Why are we calling `handleError` here?

Reply:
Since `tranformResponse` is async function, incase it throws an error we can handle it.

Reply:
Earlier, we were using `handleError` to only handle the elasticsearch query errors.

Reply:
Earlier it was Promise chain, so adding `catch` block at the end would handle it for all promises.